plugins/woocommerce/includes/rest-api/Controllers/Version3/class-wc-rest-products-controller.php

```diff
@@ -178,6 +185,10 @@ protected function prepare_objects_query( $request ) {
 			$args['post_status'] = $request['include_status'];
 		}
 
+		if ( ! empty( $request['exclude_status'] ) ) {
+			$this->exclude_status = $request['exclude_status'];
+		}
```

Comment:
🤔 I wonder if there could be a scenario where the `exclude_status` prop gets populated here by a request object, but then `get_objects` is never called so the prop never gets cleared, and then a second request object gets processed and inherits this prop from the first request object... I guess that is a pretty extreme edge case. But one way to mitigate that risk wouuld be to clear the prop here when `$request['exclude_status']` _is_ empty.

Reply:
Good catch! Updated here 31167d40417d11fedbef48dbdb2b197e7bb0bbc0